Havit/Havit/Screens/CategoryContents/View/CategoryContentsViewController.swift

```diff
@@ -6,24 +6,184 @@
 //
 
 import UIKit
+import SnapKit
+import RxSwift
 
-class CategoryContentsViewController: UIViewController {
+class CategoryContentsViewController: BaseViewController {
+    var searchController: UISearchController!
+    var mainView: UIView!
+    var filterView: UIView!
+    var totalLabel: UILabel!
+    var changeShowButton: UIButton!
+    var sortButton: UIButton!
+    
+    private var filterCollectionView: UICollectionView!
+    private var contentsCollectionView: UICollectionView!
 
     override func viewDidLoad() {
         super.viewDidLoad()
-
-        // Do any additional setup after loading the view.
+        setUI()
+        setAutoLayouts()
+        
     }
     
+    func setUI() {
+        // TODO: 네비게이션바 생성 (메인화면에서 Coordinator로 진입)
+        
+        // 검색바 생성
+        let searchController = UISearchController(searchResultsController: nil)
+        searchController.searchBar.placeholder = "원하는 콘텐츠를 검색하세요."
+        self.navigationItem.searchController = searchController
+        self.navigationItem.title = "카테고리"
+        searchController.hidesNavigationBarDuringPresentation = false
+        self.navigationItem.hidesSearchBarWhenScrolling = false
+        
+        // 메인 뷰 생성
+        mainView = UIView()
+        mainView.backgroundColor = .systemRed
+        self.view.addSubview(mainView)
+        
+        // 필터 뷰 생성
+        filterView = UIView()
+        filterView.backgroundColor = .white
+        self.view.addSubview(filterView)
+        
+        totalLabel = {
+            let label = UILabel()
+            label.text = "전체 0"
+            return label
+        }()
+        filterView.addSubview(totalLabel)
+        
+        changeShowButton = {
+            let button = UIButton()
+            button.setTitle("B", for: .normal)
+            button.backgroundColor = UIColor.blue
+            return button
+        }()
+        filterView.addSubview(changeShowButton)
+        
+        sortButton = {
+            let button = UIButton()
+            button.setTitle("최근순", for: .normal)
+            button.backgroundColor = UIColor.blue
+            return button
+        }()
+        filterView.addSubview(sortButton)
+        
+        filterCollectionView = UICollectionView(frame: CGRect.zero, collectionViewLayout: UICollectionViewFlowLayout.init())
+        filterCollectionView.backgroundColor = .blue
+        filterView.addSubview(filterCollectionView)
+        filterCollectionView.delegate = self
+        filterCollectionView.dataSource = self
+        filterCollectionView.register(CategoryFilterCollectionViewCell.classForCoder(), forCellWithReuseIdentifier: CategoryFilterCollectionViewCell.cellID)
```

Comment:
P2;
UICollectionView+Extension 파일에 CollectionViewCell register 코드를 짜주시면 감사드리겠습니다~
```suggestion
        filterCollectionView.register(cell: CategoryFilterCollectionViewCell.self)
```

Reply:
p1

Reply:
넵! 코드 이동하도록 하겠습니다!